### PR TITLE
test(network): make Eth62 tx-packing tests deterministic in size

### PR DESCRIPTION
**Pull request by AnkushinDaniil**
## Changes

- `Eth62ProtocolHandlerTests.should_send_single_transaction_even_if_exceed_MaxPacketSize(128)` failed on CI (`ubuntu-24.04-arm`, run [29806899202](https://github.com/NethermindEth/nethermind/actions/runs/29806899202/attempts/1), both in-job attempts) with "Not all expected messages were delivered within 30s", and passed on re-run.
- Root cause: each transaction is signed with a fresh random key, and RLP drops leading zero bytes of `r`/`s`, so transaction sizes vary by a byte in ~0.8 % of signatures. The expected message segmentation is derived from `txs[0].GetLength()` alone, so when `txs[0]` happens to encode shorter, the computed txs-per-message count (e.g. 455 instead of 453) no longer matches the actual greedy packing in `SendNewTransactionsCore`, no delivered message matches the assertion predicate, and the test burns the full 30-second wait before failing.
- Fix: build the transactions with a fixed full-length synthetic signature (hashes stay distinct via distinct `s` content, which the transaction hash covers), so every transaction serializes to the same length and the packing arithmetic is exact. The two sibling packing tests use the same construction and had the same latent flake, so they now share the helper.
- The `CountdownEvent`/30-second wait is removed: `SendNewTransactions` → `Session.DeliverMessage` is fully synchronous, so the wait could never rescue a mismatch — a plain `Received(...)` assertion is equivalent and fails fast.

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [x] Other: test flakiness fix

### Requires testing

- [x] Yes
- [ ] No

If yes, did you write tests?

- [x] Yes
- [ ] No

### Notes on testing

`Eth62ProtocolHandlerTests` suite (59 tests) passes locally on arm64; the three packing tests were additionally run 3× in a loop. The 1055/1056 boundary cases confirm the synthetic transactions serialize to the same length as the previous `SignedAndResolved` ones.

### Requires documentation update

- [ ] Yes
- [x] No

### Requires explanation in Release Notes

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Network.Test/P2P/Subprotocols/Eth/V62/Eth62ProtocolHandlerTests.cs [hunk 1]
```diff
@@ -765,18 +765,33 @@ static void HandlerOnSubprotocolRequested(object sender, ProtocolEventArgs e) {
...
+            r.AsSpan().Fill(0x11);
             Transaction[] txs = new Transaction[txCount];
-
             for (int i = 0; i < txCount; i++)
             {
-                txs[i] = Build.A.Transaction.SignedAndResolved(Build.A.PrivateKey.TestObject).TestObject;
+                byte[] s = new byte[32];
+                s.AsSpan().Fill(0x22);
+                s[30] = (byte)(i >> 8);
+                s[31] = (byte)i;
```
Comment (hudem1) [inline src/Nethermind/Nethermind.Network.Test/P2P/Subprotocols/Eth/V62/Eth62ProtocolHandlerTests.cs]: **Nit** Encoding the index into only the last two bytes of `s` caps hash uniqueness at 65536 txs. It's fine for the current cases (max `txCount` is 10000), but uniqueness is a load-bearing invariant here — `SendNewTransactionsCore` dedupes via `ShouldNotifyTransaction`/`NotifiedTransactions.Set`, so any hash collision silently drops txs and would make the message-count assertions under-count rather than fail loudly. If a larger `[TestCase]` is ever added this becomes a subtle trap. Consider spreading the index across more bytes (or a brief comment noting the 65535 ceiling) so the helper's contract is self-documenting.

**Review by hudem1 — APPROVED** [review]

File: src/Nethermind/Nethermind.Network.Test/P2P/Subprotocols/Eth/V62/Eth62ProtocolHandlerTests.cs (hunk 1 above)
Comment (AnkushinDaniil) [reply src/Nethermind/Nethermind.Network.Test/P2P/Subprotocols/Eth/V62/Eth62ProtocolHandlerTests.cs]: Good point, thanks! Spread the index across 4 bytes with BinaryPrimitives.WriteInt32BigEndian, so the ceiling is gone now.

**Review by wurdum — APPROVED** [review]
